### PR TITLE
Change fl.Slicing Implementation

### PR DESCRIPTION
**Pull request by limiteinductive**
The current interface for fl.Slicing currently has two flaws:
  -> It follows Pytorch's  .narrow interface, which is confusing.
  -> It cannot perform slices with implicit lengths like this `x[1:]`.
  
  In this PR, I propose to make fl.Slicing follow more closely what we could expect from a slice method with `dim`, `start`, `end`, and `step` keywords.

### CODE REVIEW

File: src/refiners/fluxion/layers/basics.py [hunk 1]
```diff
@@ -91,6 +91,8 @@ def __init__(self, dim: int, start: int, length: int) -> None:
         self.length = length
 
     def forward(self, x: Tensor) -> Tensor:
+        if self.length < 0:
+            return x.narrow(self.dim, self.start, x.shape[self.dim] - self.start + self.length)
```
Comment (deltheil) [inline src/refiners/fluxion/layers/basics.py:95]: Off-by-one error I guess?

```
>>> import refiners.fluxion.layers as fl
>>> import torch
>>> m = fl.Slicing(dim=2, start=2, length=-1)
>>> x = torch.tensor([[[0, 1, 2, 3, 4, 5, 6]]])
>>> y = m(x)
>>> z = x[:, :, 2:]
>>> y
tensor([[[2, 3, 4, 5]]])
>>> z
tensor([[[2, 3, 4, 5, 6]]])
```

**Review by deltheil — APPROVED** [review]

**Comment by Laurent2916** [comment]
I think there is still an off-by-one error:
```python
>>> import torch
>>> import refiners.fluxion.layers as fl
>>> data = torch.randn(2, 1000, 400)
>>> slice = fl.Slicing(dim=1, start=1)
>>> slice(data).shape
torch.Size([2, 998, 400])
>>> data[:, 1:, :].shape
torch.Size([2, 999, 400])
```

**Comment by limiteinductive** [comment]
> I think there is still an off-by-one error @limiteinductive:
> 
> ```python
> >>> import torch
> >>> import refiners.fluxion.layers as fl
> >>> data = torch.randn(2, 1000, 400)
> >>> slicing = fl.Slicing(dim=1, start=1)
> >>> slicing(data).shape
> torch.Size([2, 998, 400])
> >>> data[:, 1:, :].shape
> torch.Size([2, 999, 400])
> ```

Good catch! The default for the `end` param should be `None` and not `-1`. PR incoming
